beater/beater.go

```diff
@@ -29,19 +30,75 @@ func New(_ *beat.Beat, ucfg *common.Config) (beat.Beater, error) {
 	return bt, nil
 }
 
+type Eventer struct {
+	eventsLeft int
```

Comment:
test running with multiple agents send (sending events) fast and binary compiled with `-race`. Might be you want the type to be `atomic.Int32`.